tests/ui/rfcs/rfc-0000-never_patterns/check.rs

```diff
@@ -15,18 +15,18 @@ fn no_arms_or_guards(x: Void) {
         //~^ ERROR a never pattern is always unreachable
         None => {}
     }
-    match None::<Void> {
+    match None::<Void> { //~ ERROR: `Some(_)` not covered
         Some(!) if true,
```

Comment:
How is this not a parse error!?

Reply:
It is, but we fall back to generating all arms up to and including the parse error arm.

Reply:
I tried replacing the entire match with an error expression, but then we lose the information that we don't need a semicolon after it. I guess I could encode that fact in the error variant. I'll try that out tomorrow

Reply:
I was just surprised that we didn't have an `//~ERROR` annotation in the test.